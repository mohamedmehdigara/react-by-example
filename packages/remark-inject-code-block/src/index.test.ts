import { promises } from 'fs'
import { extname } from 'path'
import remark from 'remark'
import { Processor } from 'unified'
import remarkInjectCodeBlock, { Resolved } from '.'

const { readFile } = promises

const fixtures = {
  simple: `\
# Hello world!

## Heading level 2

![](@app/example-packages/example-packages/hello-world/src/index.tsx)
![](@app/example-packages/example-packages/simple-clock/public/index.html)
`,
}

describe('remarkInjectCodeBlock', () => {
  let processor: Processor

  beforeEach(() => {
    processor = remark().use(remarkInjectCodeBlock, {
      async resolve({ type, url }): Promise<Resolved> {
        if (type !== 'image') return undefined
        if (typeof url !== 'string') return undefined

        const path = require.resolve(url)
        const lang = extname(path).slice(1)
        const code = await readFile(path, 'utf8')
        return { code, lang }
      },
    })
  })

  it.each(Object.entries(fixtures))(
    'should inject code blocks "%s"',
    async (_name, markdown) => {
      const result = await processor.process(markdown)
      expect(result.toString()).toMatchSnapshot()
    },
  )
})
