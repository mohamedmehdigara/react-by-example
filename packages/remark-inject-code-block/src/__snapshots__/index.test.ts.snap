// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remarkInjectCodeBlock should inject code blocks "simple" 1`] = `
"# Hello world!

## Heading level 2

\`\`\`tsx
import React from 'react'
import ReactDOM from 'react-dom'

ReactDOM.render(<h1>Hello world!</h1>, document.getElementById('root'))
\`\`\`![](@app/example-packages/example-packages/hello-world/src/index.tsx)
\`\`\`html
<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\" />
    <title>React App</title>
  </head>
  <body>
    <noscript>You need to enable JavaScript to run this app.</noscript>
    <div id=\\"root\\"></div>
  </body>
</html>
\`\`\`![](@app/example-packages/example-packages/simple-clock/public/index.html)
"
`;
